public/app/plugins/panel/table/transformers.ts

```diff
@@ -135,19 +135,133 @@ transformers['table'] = {
     if (!data || data.length === 0) {
       return [];
     }
-    return data[0].columns;
+
+    // Single query returns data columns as is
+    if (data.length === 1) {
+      return [...data[0].columns];
+    }
+
+    // Track column indexes: name -> index
+    const columnNames = {};
+
+    // Union of all non-value columns
+    const columns = data.reduce((acc, d, i) => {
+      d.columns.forEach((col, j) => {
+        const { text } = col;
+        if (text !== 'Value') {
+          if (columnNames[text] === undefined) {
+            columnNames[text] = acc.length;
+            acc.push(col);
+          }
+        }
+      });
+      return acc;
+    }, []);
+
+    // Append one value column per data set
+    data.forEach((_, i) => {
```

Comment:
Sorry for the scattered feedback... too many things going on today. 

This assumes that the table data have a value field. Which is true for Prometheus but not for InfluxDb / Elastic

Reply:
No worries. It's a complex software. 
Let's revisit the essence of the PR: producing a union of columns and merging rows that have the same non-metric field values.
I see a couple of ways forward: 

a) probe the first rows of each series for which types are metric fields and adjust the merge semantic accordingly (I do realize that relying on 'Value' labels was brittle to begin with)
b) keep ignoring the additional queries for data sources other than prometheus.
c) accept the undefined behavior for non-prometheus sources and merge as is

@bergquist or do you see other options? Let me know what you prefer...

Reply:
B) Feels unconsistent. Really want to avoid such things when we have so many data sources. 
C) This would be fine if the code change would be contained in the Prometheus plugin. But since it's in a panel I don't like the idea. 
A) I wonder how good this would workout. How would be decide if the should merge or create a new column? strings vs numbers would not be good enough for elastic/influxdb. 

Another approach would be to ask the Prometheus data source to give columns a unique name when returning multiple datasets as a table. ex: https://github.com/bergquist/grafana/commit/962709a2db9c3a1ae95940c3c1a1b13a57d117f9

Reply:
I like the renaming of the value column in the prom datasource, that simplifies the column union.
For the merge/join logic, if we cant do probing based on data types, we could simply merge on missing indexes in the shifted rows, e.g., row 1 can merge with a row that has missing values on `index(V2)` while other values must match unless they are undefined:
```
   C1 | V1 | V2 | TS
1. foo| 123| -  | 1
2. foo|  - | 454| 1
3. bar|  - | 890| 1
```
I'll have a think.